src/io/network/endpoint.cpp

```diff
@@ -109,4 +145,78 @@ std::ostream &operator<<(std::ostream &os, const Endpoint &endpoint) {
   return os << endpoint.address << ":" << endpoint.port;
 }
 
+bool Endpoint::IsResolvableAddress(const std::string &address, uint16_t port) {
+  addrinfo hints{
+      .ai_flags = AI_PASSIVE,
+      .ai_family = AF_UNSPEC,     // IPv4 and IPv6
+      .ai_socktype = SOCK_STREAM  // TCP socket
+  };
+  addrinfo *info = nullptr;
+  auto status = getaddrinfo(address.c_str(), std::to_string(port).c_str(), &hints, &info);
+  return status == 0;
+}
+
+std::optional<std::pair<std::string, uint16_t>> Endpoint::ParseSocketOrAddress(
+    const std::string &address, const std::optional<uint16_t> default_port = {}) {
+  const std::string delimiter = ":";
+  std::vector<std::string> parts = utils::Split(address, delimiter);
+  if (parts.size() == 1) {
+    if (GetIpFamily(address) == IpFamily::NONE) {
+      if (!IsResolvableAddress(address, *default_port)) {
+        return std::nullopt;
+      }
+      return ParseHostname(address, default_port);
```

Comment:
We do not want to prematurely resolve to an ip address.
We only want to know that ATM it is resolvable, we only want to split off the address from the port. 
Reason for this is this address is persisted, which means at server restart we want the url to be recovered so we can re-resolve the ip address.
DNS entries may be different on restart.